pkg/os/shell/shell_windows.go

```diff
@@ -67,32 +91,18 @@ func detect() (string, error) {
 		if err != nil {
 			return "cmd", err // defaulting to cmd
 		}
-		switch {
-		case strings.Contains(strings.ToLower(shell), "powershell"):
-			return "powershell", nil
-		case strings.Contains(strings.ToLower(shell), "pwsh"):
-			return "powershell", nil
-		case strings.Contains(strings.ToLower(shell), "cmd"):
-			return "cmd", nil
-		default:
-			shell, _, err := getNameAndItsPpid(shellppid)
-			if err != nil {
-				return "cmd", err // defaulting to cmd
-			}
-			switch {
-			case strings.Contains(strings.ToLower(shell), "powershell"):
-				return "powershell", nil
-			case strings.Contains(strings.ToLower(shell), "cmd"):
-				return "cmd", nil
-			default:
-				return "cmd", nil // this could be either powershell or cmd, defaulting to cmd
-			}
-		}
+		return analyzeShell(shell, shellppid)
 	}
 
 	if os.Getenv("__fish_bin_dir") != "" {
 		return "fish", nil
 	}
 
-	return filepath.Base(shell), nil
+	baseShell := filepath.Base(shell)
+
+	if baseShell != "powershell" && baseShell != "cmd" {
```

Comment:
At this point, `shell` was non-empty, but it was neither `powershell` nor `cmd`, so we need to do some matching on the name of the parent process? Some explanations in the commit log about what the expectations are could be useful.
`analyzeShell` does the right thing if `shell` is `powershell` or `cmd`, so you could just replace `return filepath.Base(shell), nil` in the initial code with `return analyzeShell(shell, pid)`, no need for `baseShell` I think.

Reply:
Looks clear, yes